src/components/Header.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { MakeChange } from '../types/MakeChange';
+
+type Props = {
+  setTodos: MakeChange,
+};
+
+export const Header: React.FC<Props> = ({
+  setTodos,
+}) => {
+  const [value, setValue] = useState('');
+  const input = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    input.current?.focus();
+  }, []);
+
+  const hendlerInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+  };
+
+  const hendlerForm = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (value.trim()) {
+      setTodos.add({
```

Comment:
Instead of using 'Date.now()' for the 'id', consider using a more unique id generation strategy, such as 'uuid' library